### DISCUSSION TITLE
Check if function return type is an interface

### DISCUSSION THREAD

**Discussion by rvigo** [LemonAppDev/konsist#824] [Ask a Question] [answered]
### Wanted end result.

having this interface
``` kotlin
interface Person {
    val name: String
}
```

and another class with this method
``` kotlin
fun getPerson() : Person {
    //...
}
```
I want to check if the return type (in this case, `Person`) is an interface.

so, in the example above, it should return `true` and for an implementation, should return `false`

### What you tried so far.

I got this (for an example where I want to check if the interfaces in a package has another interface as return type):

``` kotlin
  scope
            .interfaces()
            .functions()
            .withReturnValue()
            .returnTypes
            .assertTrue {
                // this is the last info I can get
                it.sourceType 
            }
```

is there a way to check the modifier the same way we can check if the class is abstract?

**Comment by PoisonedYouth** [comment]
I had a similar request and found a workaround. Maybe this can help you temporarily: https://kotlinlang.slack.com/archives/C05QG9FD6KS/p1705999291070539?thread_ts=1705999291.070539&cid=C05QG9FD6KS

**Comment by nataliapeterwas** [comment, marked as answer]
This will be possible in the next release which will be available soon.

Upd (link to PR):
https://github.com/LemonAppDev/konsist/pull/868

**Comment by nataliapeterwas** [comment]
A new version v0.14.0 has been released. It contains the fix.
See [v0.14.0 release](https://github.com/LemonAppDev/konsist/releases/tag/v0.14.0)

Solution:
```kotlin
scope
	.interfaces()
	.functions()
	.withReturnValue()
	.returnTypes
	.assertTrue { it.isInterface }
```

**Reply by rvigo** [reply]
thanks so much @nataliapeterwas 🚀
